### ISSUE TITLE
[Feature request] Allow decorate to take optional override names for properties

### ISSUE THREAD

**Issue by natew** [mobxjs/mobx#1603]
3. [x] Idea:
  * [x] What problem would it solve for you?

In the docs example:

```js
class TodoList {
    todos = {}
    get unfinishedTodoCount() {
        return values(this.todos).filter(todo => !todo.finished).length
    }
    addTodo() {
        const t = new Todo()
        t.title = 'Test_' + Math.random()
        set(this.todos, t.id, t)
    }
}

decorate(TodoList, {
    todos: observable,
    unfinishedTodoCount: computed,
    addTodo: action.bound
})
```

When you attach Mobx debugger or Mobx.trace() to this you'll see something like `addTodo` over and over. I'd prefer to have it show `TodoList.addTodo` and maybe even in my own case attach some instance level stuff like `SomeParent.TodoList.addTodo`.

* [x] Are you willing to (attempt) a PR yourself?

Sure thing.

Basically would propose having something like:

```js
class TodoList {
    todos = {}
    get unfinishedTodoCount() {
        return values(this.todos).filter(todo => !todo.finished).length
    }
    addTodo() {
        const t = new Todo()
        t.title = 'Test_' + Math.random()
        set(this.todos, t.id, t)
    }
}

decorate(TodoList, {
    todos: observable,
    unfinishedTodoCount: computed,
    addTodo: action.bound
}, {
  todos: 'TodoList.todos',
})
```

Where that would name it as such.

**Comment by mweststrate** [comment]
I see why this could be useful, but I think it is rare enough to not further complicate the api. Closing for now, until more need arises
